src/app/static/src/app/components/outputTable/TableFilters.vue

```diff
@@ -1,3 +1,134 @@
+import { PlottingSelectionsMutations } from "../../store/plottingSelections/mutations";
+import { TableSelections } from "../../store/plottingSelections/plottingSelections";
+import { CalibrateMetadataResponse, FilterOption } from "../../generated";
+import NewFilters from "./NewFilters.vue";
+import { SingleSelect } from "@reside-ic/vue-nested-multiselect";
+
+type Preset = CalibrateMetadataResponse["tableMetadata"]["presets"][number];
+
+export default defineComponent({
+    components: {
+        NewFilters,
+        SingleSelect
+    },
+    setup() {
+        const store = useStore<RootState>();
+        const tableSelections = computed(() => store.state.plottingSelections.table);
+        const presets = computed(() => store.state.modelCalibrate.metadata?.tableMetadata.presets);
+        const filters = computed<DisplayFilter[]>(() => store.getters["modelOutput/tableFilters"] || []);
+
+        const presetOptions = computed(() => {
+            return presets.value!.map(p => {
+                return { id: p.defaults.id, label: p.defaults.label }
+            });
+        });
+
+        const indicatorOptions = computed(() => {
+            const indicators = store.state.modelCalibrate.metadata?.plottingMetadata.choropleth.indicators;
+            return indicators?.map(indicator => {
+                return { id: indicator.indicator, label: indicator.name };
+            }) || [];
+        });
+        
+        const updateTableSelections = (payload: Partial<TableSelections>) => {
+            store.commit(
+                `plottingSelections/${PlottingSelectionsMutations.updateTableSelections}`,
+                { payload }
+            );
+        };
+
+        const changeFilterSelections = (selections: Dict<FilterOption[]>) => {
+            updateTableSelections({ selectedFilterOptions: selections });
+        };
+
+        const changeIndicatorSelection = (indicatorOption: Partial<FilterOption>) => {
+            updateTableSelections({ indicator: indicatorOption.id });
+        };
+
+        const toFilterOption = (filterId: string, optionId: string) => {
+            const filter = filters.value.find(f => f.id === filterId);
+            return filter?.options.find(option => option.id === optionId);
+        };
+
+        const getDefaultSelections = (preset: Preset) => {
+            const defaultSels = Object.fromEntries(filters.value.map(f => {
+                if (f.id === preset.defaults.column || f.id === preset.defaults.row) {
+                    return [f.id, f.options]
+                }
+                return [f.id, [f.options[0]]]
+            }));
+            return defaultSels
+        };
+
+        const changePresetSelection = (presetOption: Partial<FilterOption>) => {
+            const currentPreset = presets.value?.find(p => p.defaults.id === presetOption.id);
+            if (!currentPreset) return;
+
+            updateTableSelections({ preset: presetOption.id });
```

Comment:
So this is because the filters can be different for different presets, first we have to commit the preset to get the new table filters and then the second commit for the selections once we have the new filters